pkg/analysis/optionalfields/analyzer.go

```diff
@@ -303,9 +364,15 @@ func getStructZeroValue(pass *analysis.Pass, structType *ast.StructType) string
 	for _, field := range structType.Fields.List {
 		fieldTagInfo := jsonTagInfo.FieldTags(field)
 
+		isPointer, _ := isStarExpr(field.Type)
+		if !isPointer && fieldTagInfo.OmitZero {
+			// for non-pointer field if it has omitzero, we can use a zero value.
```

Comment:
I'm not following this

Reply:
I added this check based on earlier review comment https://github.com/kubernetes-sigs/kube-api-linter/pull/115#discussion_r2189495095, Did I misunderstood?

Reply:
I don't get why we say `we can use a zero value`?

Non-pointer and omitzero is an acceptable state, but it means that the zero value is not valid, so your comment is contradicting this no?

Reply:
apologies for the confusion, What I try to convey is that field is non-pointer and has omitzero then the field will be omitted and we can consider it as a zero value.

Reply:
Updated it, Please let me know if it still incorrect.